share/eds/store.go

```diff
@@ -164,22 +168,30 @@ func (s *Store) gc(ctx context.Context) {
 // The square is verified on the Exchange level, and Put only stores the square, trusting it.
 // The resulting file stores all the shares and NMT Merkle Proofs of the EDS.
 // Additionally, the file gets indexed s.t. store.Blockstore can access them.
-func (s *Store) Put(ctx context.Context, root share.DataHash, square *rsmt2d.ExtendedDataSquare) (err error) {
-	// if root already exists, short-circuit
-	has, err := s.Has(ctx, root)
-	if err != nil {
-		return fmt.Errorf("failed to check if root already exists in index: %w", err)
-	}
-	if has {
-		return dagstore.ErrShardExists
-	}
-
+func (s *Store) Put(ctx context.Context, root share.DataHash, square *rsmt2d.ExtendedDataSquare) error {
 	ctx, span := tracer.Start(ctx, "store/put", trace.WithAttributes(
 		attribute.Int("width", int(square.Width())),
 	))
-	defer func() {
-		utils.SetStatusAndEnd(span, err)
-	}()
+
+	tnow := time.Now()
+	err := s.put(ctx, root, square)
+	switch {
+	case errors.Is(err, dagstore.ErrShardExists):
+		s.metrics.observePut(ctx, time.Since(tnow), putExists, square.Width())
+	case err != nil:
+		s.metrics.observePut(ctx, time.Since(tnow), putFailed, square.Width())
+	default:
+		s.metrics.observePut(ctx, time.Since(tnow), putOK, square.Width())
+	}
+	utils.SetStatusAndEnd(span, err)
```

Comment:
Readability nit
```suggestion

  result := putOk 
	tnow := time.Now()
	err := s.put(ctx, root, square)
	switch {
	case errors.Is(err, dagstore.ErrShardExists):
		result = putExists
	case err != nil:
		result = putFailed
	}
	utils.SetStatusAndEnd(span, err)
	s.metrics.observePut(ctx, time.Since(tnow), putOK, square.Width())
```

Reply:
Applied. Had to do manually, since you last line of changes needs to use result

Reply:
All good nw